packages/block-editor/src/components/block-list/use-block-props/use-event-handlers.js

```diff
@@ -125,17 +120,23 @@ export function useEventHandlers( clientId ) {
 				event.preventDefault();
 			}
 
-			node.addEventListener( 'keydown', onKeyDown );
-			node.addEventListener( 'mouseleave', onMouseLeave );
-			node.addEventListener( 'dragstart', onDragStart );
+			node.addEventListener( 'focusin', onFocus );
```

Comment:
Why would we have to attach an event handler to select a block if the block is already selected? Are we too late to attach the event handler in case a block is unselected? Or what's the root issues here?

Reply:
@ellatrix You're right, I think have mis-diagnosed this. 

The problem is most easily repro'd in the Navigation Editor (though I've also managed to reproduce it in the post editor, but it seems harder)
1. Create a new menu
2. Click on the block and choose the 'Start empty' option
3. Deselect the block
4. Try selecting the block again - although it receives focus it isn't selected

The main problem seems to be that the `useRefEffect` callback in `useEventHandlers` isn't triggered when deselecting the block (the `isSelect` dependency flips from `true` to `false`, so it should be triggered at that point):
https://github.com/WordPress/gutenberg/blob/6be732011f9741e13dd248c64eea33c9dd76aca7/packages/block-editor/src/components/block-list/use-block-props/use-event-handlers.js#L50-L137

The result is the event handler isn't bound and the block can't be selected.

Reply:
Thanks. I'll have a look into it now.

Reply:
Fixed in #29892.